src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1560,6 +1586,98 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
+                    // indicates that we have e.g., some region variable
+                    // whose value is like `'a+'b` where `'a` and `'b` are
+                    // distinct unrelated univesal regions that are not
+                    // known to outlive one another. It'd be nice to have
+                    // some examples where this arises to decide how best
+                    // to report it; we could probably handle it by
+                    // iterating over the universal regions and reporting
+                    // an error that multiple bounds are required.
+                   self.infcx.tcx.sess
+                       .struct_span_err(
+                           span,
+                           &format!("`{}` does not live long enough", generic),
+                       )
+                       .buffer(&mut self.errors_buffer);
+                },
+
+                RegionErrorKind::TypeTestGenericBoundError {
+                    span, generic, lower_bound_region
+                } => {
+                    let region_scope_tree = &self.infcx.tcx.region_scope_tree(self.mir_def_id);
+                    self.infcx
+                        .construct_generic_bound_failure(
+                            region_scope_tree,
+                            span,
+                            None,
+                            generic,
+                            lower_bound_region,
+                        )
+                        .buffer(&mut self.errors_buffer);
+                },
+
+                RegionErrorKind::UnexpectedHiddenRegion {
+                    opaque_type_def_id, hidden_ty, member_region,
+                } => {
+                    let region_scope_tree = &self.infcx.tcx.region_scope_tree(self.mir_def_id);
+                    opaque_types::unexpected_hidden_region_diagnostic(
+                        self.infcx.tcx,
+                        Some(region_scope_tree),
+                        opaque_type_def_id,
+                        hidden_ty,
+                        member_region,
+                    )
+                    .buffer(&mut self.errors_buffer);
+                }
+
+                RegionErrorKind::RegionError {
+                    fr_origin, longer_fr, shorter_fr,
+                } => {
+                    let db = self.nonlexical_regioncx.report_error(
+                        self,
+                        longer_fr,
+                        fr_origin,
+                        shorter_fr,
+                        &mut outlives_suggestion,
+                        &mut region_naming,
+                    );
+
+                    db.buffer(&mut self.errors_buffer);
+                }
+
+                RegionErrorKind::UnreportedError {
+                    longer_fr, shorter_fr,
+                    fr_origin: _,
+                } => {
+                    // FIXME: currently we do nothing with these, but perhaps we can do better?
```

Comment:
Wait, these are errors that we don't report...? Shouldn't there be like a `.delay_span_bug` at least or something?

Reply:
No, they are intentionally never reported, as mentioned in the comments

Reply:
Oh I see, by "the" you mean:

    /// The same as `RegionError`, but this is an unreported error. We currently only report the
    /// first error encountered and leave the rest unreported so as not to overwhelm the user.

Maybe leave a short note here as well? (The comment is slightly terrifying atm when you don't know the comment from the variant.)

Reply:
@Centril I added a comment in https://github.com/rust-lang/rust/pull/67241